frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -338,6 +338,28 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        #[pallet::weight(T::BlockWeights::get().max_block / 5 * 3)]
+        pub fn do_upgrade(
+            origin: OriginFor<T>,
+            weight_limit: Option<Weight>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
```

Comment:
shouldn't this be done by root or sudo?

Reply:
No, we set it intentionally so anyone can do it.
This makes it easier to do calls since e.g. I can do it without sudo access.

They way call is written, it's very expensive so it cannot be spammed.
Also, there are safeguards in place which basically make the call obsolete once migration is finished.

Reply:
ah, got it, thanks for the explanation 👍